WordPress/src/main/java/org/wordpress/android/ui/main/WPMainActivity.java

```diff
@@ -724,8 +725,10 @@ public void onAuthenticationChanged(OnAuthenticationChanged event) {
 
             if (mIsMagicLinkLogin) {
                 if (mIsMagicLinkSignup) {
-                    mLoginAnalyticsListener.trackCreatedAccount();
-                    mLoginAnalyticsListener.trackSignupMagicLinkSucceeded();
+                    mDispatcher.dispatch(AccountActionBuilder.newFetchAccountAction());
+                    // While unlikely, its possible the app could be closed before the onAccountChanged event is
+                    // handled. Setting this flag lets us dispatch the event the next time the app opens.
```

Comment:
Setting this flag doesn't actually lets us dispatch the event, it makes sure it's always handled in the `OnAccountChanged` even if the user quits the app after signup is completed and before `OnAccountChanged` could be triggered. Feel free to move this comment under `onAccountChanged` (or the helper method I mentioned in the other comment) if you think it'll be easier to understand where it's used.

Reply:
I'll move it to the helper